### DISCUSSION TITLE
Is there a way to tell Weblate that an "Unchanged translation" is ok for a particular string?

### DISCUSSION THREAD

**Discussion by luebbe** [WeblateOrg/weblate#5738] [Q&A]
Quite often there are terms that stay the same accross languages. An example is the abbreviation "hex" which is translated to "hex" in German. A lot of technical terms are affected.

I tried adding this pair to the glossary, but the check still fails.
When I discard the warning for one component, it is still visible in other components, and it will probably also show up for any new component.

I see there is a blacklist for unchanged strings, but I didn't find information where this can be edited. Is this blacklist per language?

**Comment by nijel** [comment]
The blacklist is in the source code and is shared for all languages.

**Reply by luebbe** [reply]
Thanks, I thought so. So the blacklist wouldn't help in this case.

**Comment by luebbe** [comment]
As far as I understand glossary/terminology contains a definition how the term "has to" be translated into a certain language.

Would it make sense to either:
- skip the "unchanged translation" check for terms that are in a glossary?
- or, if that is too general, allow to add a "skip-unchanged" flag to glossary terms, so that the check can be skipped selectively?

**Reply by nijel** [reply]
The glossary is currently not included in the default set of checks for performance reasons (see https://docs.weblate.org/en/latest/user/checks.html#does-not-follow-glossary), so including it here is problematic for the same reasons as well, but it probably could be addressed for this part (it would need only non-translatable strings from the glossary, what will be smaller scale than looking up all glossary terms).

**Comment by nijel** [comment]
If these are technical terms which are not translated into more languages, then it makes sense to include them in the built-in blacklist. Most of the existing terms were collected based on analysis of failed checks on Hosted Weblate this way.

**Reply by luebbe** [reply]
In my case probably not. Most of the terms are particular to a certain industry. "hex" is just a simple example.

Another example are single words like "Routine" on the ui:
de - Routine
fr - Routine
es - Rutina
ja - ルーチン
it - Routine
...

This should never make it into a central blacklist, because it has to be translated in many cases (unless I understand the blacklist wrong).
These terms occur in several components, so it would be nice to be able to turn the check off for them per language in a central location.

**Reply by nijel** [reply]
The list should contain terms which are often not translated. It is okay to miss some unchanged translations. See https://github.com/WeblateOrg/weblate/blob/main/weblate/checks/data.py for current one.

**Reply by nijel** [reply]
..and actually routine is already there

**Reply by luebbe** [reply]
> 
> 
> ..and actually routine is already there

This is funny, because the concrete labels are called "routine 1, routine 2, ...", so the blacklist doesn't catch them. Nevermind :)
I still think that a global attribute to turn off the check per term and language would be helpful.

**Reply by nijel** [reply]
Can you please share the exact configuration where you have this issue? I've added a test for this string in 69d7ef01a08869f866d5623406ff0ac780cb659e, and it doesn't fire the check.

Do you happen to use `strict-same` flag? See https://docs.weblate.org/en/latest/user/checks.html#check-same

**Reply by luebbe** [reply]
@nijel sorry, my fault. The labels are "routine1, routine2, ..." without blank. 

`strict-same` probably not. Currently I'm testing a lot by creating new projects and components from scratch via the api and deleting them before I test again. So everything should be in "default" mode.

**Reply by nijel** [reply]
You can check the flag in the string information pane while translating.

It is turned on by default some document formats, what format do you use?

**Reply by luebbe** [reply]
No, it wasn't on.

.po files and some .islu
